### PR TITLE
chore: Add `dialog` example

### PR DESCRIPTION
**Pull request by saideepesh000**

### CODE REVIEW

**Comment by saideepesh000** [comment]
@diegohaz , let me know what I can change/improve

File: packages/ariakit/src/dialog/__examples__/dialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export default function Example() {
+  const dialog = useDialogState();
+  return (
+    <>
+      <DialogDisclosure className="button" state={dialog}>
+        Open dialog
+      </DialogDisclosure>
+      <Dialog className="dialog" state={dialog} aria-label="Welcome">
+        Welcome to Ariakit!
+      </Dialog>
```
Comment (diegohaz) [inline packages/ariakit/src/dialog/__examples__/dialog/index.tsx]:
```suggestion
      <DialogDisclosure state={dialog} className="button">
        View details
      </DialogDisclosure>
      <Dialog state={dialog} className="dialog">
        <DialogDismiss className="dismiss" />
        <DialogHeading className="heading">Apples</DialogHeading>
        <ul>
          <li><strong>Calories:</strong> 95</li>
          <li><strong>Carbs:</strong> 25 grams</li>
          <li><strong>Fibers:</strong> 4 grams</li>
          <li><strong>Vitamin C:</strong> 14% of the Reference Daily Intake (RDI)</li>
          <li><strong>Potassium:</strong> 6% of the RDI</li>
          <li><strong>Vitamin K:</strong> 5% of the RDI</li>
        </ul>
      </Dialog>
```

**Review by diegohaz — COMMENTED** [review]
Thank you @saideepesh000!

I think we should have a more real-world example. I'm not really sure what we should do here though. The most common use cases for dialogs have forms. If we go that way, we would have a quite complex example using the `Dialog` and the `Form` components, which I don't think is ideal for a default example, since people will be there to learn how to use `Dialog`, and not `Form`.

Other use cases are "details" dialogs, such as product or gallery lightboxes. Those are simpler. I made a suggestion based on that, but feel free to propose a different thing.

Besides `DialogDisclosure` and `Dialog`, we also need to use `DialogHeading` and `DialogDismiss`, because it will result in better accessibility (a visible label and a visible close button).

Also pinging @leonardoelias who might have other ideas.

**Comment by saideepesh000** [comment]
I'll update this ASAP,  not getting enough time due to some other works

**Review by diegohaz — APPROVED** [review]
Thanks for starting this @saideepesh000! I updated the styles and added more tests. I think it's good to go for now. 🚀

**Comment by saideepesh000** [comment]
> Thanks for starting this @saideepesh000! I updated the styles and added more tests. I think it's good to go for now. 🚀

Hey @diegohaz , thanks for taking this forward, I'm so sorry for not wrapping this up, I was actually looking for a switch and started preparing and attending interviews so rigorously for the past couple of months. 

But burnout now, taking some time to rejuvenate myself, but I'll continue working on this very soon 😊.

**Comment by diegohaz** [comment]
> > Thanks for starting this @saideepesh000! I updated the styles and added more tests. I think it's good to go for now. 🚀
> 
> 
> 
> Hey @diegohaz , thanks for taking this forward, I'm so sorry for not wrapping this up, I was actually looking for a switch and started preparing and attending interviews so rigorously for the past couple of months. 
> 
> 
> 
> But burnout now, taking some time to rejuvenate myself, but I'll continue working on this very soon 😊.

No worries. You've done a great job. Take your time. You'll be always welcome here.

**Pull request merged**